### PR TITLE
Only use async pre-empt hack if go < 1.15

### PR DESCRIPTION
**Pull request by zeripath**
Go 1.15 should have fixed the problems requiring setting GODEBUG=asyncpreempt=off. Therefore if we are running 1.15 we don't need to set this.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: modules/git/command.go [hunk 1]
```diff
@@ -130,7 +132,9 @@ func (c *Command) RunInDirTimeoutEnvFullPipelineFunc(env []string, timeout time.
 	}
 
 	// TODO: verify if this is still needed in golang 1.15
```
Comment (techknowlogick) [inline modules/git/command.go:134]:
```suggestion
```

Comment (zeripath) [reply modules/git/command.go:134]: perhaps we should actually verify it first... :stuck_out_tongue:

File: modules/git/command.go [hunk 2]
```diff
@@ -11,10 +11,12 @@ import (
 	"io"
 	"os"
 	"os/exec"
+	"runtime"
 	"strings"
 	"time"
 
 	"code.gitea.io/gitea/modules/process"
+	"github.com/mcuadros/go-version"
```
Comment (6543) [inline modules/git/command.go:19]: github.com/mcuadros/go-version was archived !!!

we should switch to `github.com/hashicorp/go-version`

Comment (6543) [reply modules/git/command.go:19]: I'll try to create a pr - it shold be a drop in replacement ...

**Review by 6543 — APPROVED** [review]
I wont block things because #12719 change more code as i thought and is WIP at the moment

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**